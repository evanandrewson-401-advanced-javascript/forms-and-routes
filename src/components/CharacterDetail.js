import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, photoUrl, gender, affiliation, eye, hair, skin, love, weapon, profession, position, allies, enemies }) => {
  let alliesElements;
  let enemiesElements;
  if(allies) {
    alliesElements = allies.map((ally, i) => {
      return <li key={i}>{ally}</li>;
    });
  }
  if(enemies) {
    enemiesElements = enemies.map((enemy, i) => {
      return <li key={i}>{enemy}</li>;
    });
  }
  return (
    <div>
      {photoUrl && <img src={photoUrl} />}
      <h1>{name}</h1>
      {gender && <p>Gender: {gender}</p>}
      {eye && <p>Eyes: {eye}</p>}
      {skin && <p>Skin: {skin}</p>}
      {hair && <p>Hair: {hair}</p>}
      {weapon && <p>Weapon: {weapon}</p>}
      {profession && <p>Profession: {profession}</p>}
      {position && <p>Position: {position}</p>}
      {affiliation && <p>Affiliation: {affiliation}</p>}
      {allies && <ul>Allies:
        {alliesElements}
      </ul>}
      {enemies && <ul>Enemies:
        {enemiesElements}
      </ul>}
      {love && <p>Love: {love}</p>}

    </div>
  );
};

CharacterDetail.propTypes = {
  name: PropTypes.string,
  photoUrl: PropTypes.string,
  gender: PropTypes.string,
  affiliation: PropTypes.string,
  eye: PropTypes.string,
  hair: PropTypes.string,
  skin: PropTypes.string,
  love: PropTypes.string,
  weapon: PropTypes.string,
  profession: PropTypes.string,
  position: PropTypes.string,
  allies: PropTypes.array,
  enemies: PropTypes.array
};

export default CharacterDetail;
