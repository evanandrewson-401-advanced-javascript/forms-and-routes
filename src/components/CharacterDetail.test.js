import CharacterDetail from './CharacterDetail';
import React from 'react';
import { shallow } from 'enzyme';

const character = {
  _id: '5cf5679a915ecad153ab6a45',
  allies: [
    'Hakoda'
  ],
  enemies: [
    'Ozai'
  ],
  photoUrl: 'https://vignette.wikia.nocookie.net/avatar/images/c/cc/Sokka.png/revision/latest?cb=20140905085428',
  name: 'Sokka',
  gender: 'Male',
  eye: 'Blue',
  hair: 'Dark brown ',
  skin: 'Brown',
  love: ' Kaya (love interest; formerly) Suki (girlfriend) Yue (girlfriend; deceased; reincarnated as the moon spirit)',
  weapon: 'Sokka\'s weapons',
  profession: ' Politician Tactician Warrior',
  position: ' Chief of Southern Water Tribe (formerly) Map reader Son of Head Chieftain Hakoda Southern Water Tribe representative and Chairman of the United Republic Council (formerly)',
  affiliation: ' Bendless Boomerang Club[7] Southern Water Tribe Team Avatar United Republic Council',
  first: '',
  __v: 0,
};

describe('Character Detail', () => {
  it('renders a detailed Character', () => {
    const wrapper = shallow(<CharacterDetail {...character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
