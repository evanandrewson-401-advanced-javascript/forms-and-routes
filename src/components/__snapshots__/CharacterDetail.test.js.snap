// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail renders a detailed Character 1`] = `
<div>
  <img
    src="https://vignette.wikia.nocookie.net/avatar/images/c/cc/Sokka.png/revision/latest?cb=20140905085428"
  />
  <h1>
    Sokka
  </h1>
  <p>
    Gender: 
    Male
  </p>
  <p>
    Eyes: 
    Blue
  </p>
  <p>
    Skin: 
    Brown
  </p>
  <p>
    Hair: 
    Dark brown 
  </p>
  <p>
    Weapon: 
    Sokka's weapons
  </p>
  <p>
    Profession: 
     Politician Tactician Warrior
  </p>
  <p>
    Position: 
     Chief of Southern Water Tribe (formerly) Map reader Son of Head Chieftain Hakoda Southern Water Tribe representative and Chairman of the United Republic Council (formerly)
  </p>
  <p>
    Affiliation: 
     Bendless Boomerang Club[7] Southern Water Tribe Team Avatar United Republic Council
  </p>
  <ul>
    Allies:
    <li
      key="0"
    >
      Hakoda
    </li>
  </ul>
  <ul>
    Enemies:
    <li
      key="0"
    >
      Ozai
    </li>
  </ul>
  <p>
    Love: 
     Kaya (love interest; formerly) Suki (girlfriend) Yue (girlfriend; deceased; reincarnated as the moon spirit)
  </p>
</div>
`;
